docs/resources/featurestudio.md

```diff
@@ -0,0 +1,234 @@
+
+- `qAdjacent`
+- `qSubtraction` / `qUnion`
+- `qNthElement`
+- `areQueriesEquivalent`
+
+All of their documentation/use cases are available [here](https://cad.onshape.com/FsDoc);
+
+### IDs
+Every operation in Feature Studio must have a unique ID. The `id` variable is given to your featurescript at runtime that you should append before each assigned ID. ID's are stored hierarchically and each routine should have a unique ID prefix. The `+` operator tells Onshape to make a new level. Take for instance the following pseudocode:
+
+```ts
+// create a sketch with ID `rect1ID`
+const rect1ID = id+"r"+1
+createRect(rect1ID); 
+
+// extrude the sketch identified by `rect1ID`
+const extrude1ID = id+"e"+1
+extrude(extrude1ID, rect1ID); 
+```
+This creates a valid hierarchy. Each parent element is comprised of unique children. When Onshape executes this, it looks like this:
+
+<center markdown>![](../images/featurestudio/featurestudio-ex1.gif){width=80%}</center>
+
+Now, lets look at an example of what an invalid hierarchy would look like.
+
+```ts
+// create a sketch with ID `rect1ID`
+const rect1ID = id+"r"+1
+createRect(rect1ID); 
+
+// extrude the sketch identified by `rect1ID`
+const extrude1ID = id+"e"+1
+extrude(extrude1ID, rect1ID); 
+
+// create a sketch with ID `rect2ID`
+const rect2ID = id+"r"+2
+createRect(rect2ID); 
+
+// extrude the sketch identified by `rect2ID`
+const extrude2ID = id+"e"+2
+extrude(extrude2ID, rect2ID); 
+```
+
+It's hierarchy looks like this.
+
+<center markdown>![](../images/featurestudio/featurestudio-ex2.gif){width=80%}</center>
+
+Notice how there the order of creation is id->r, id->e, id->r. Onshape does not allow two nonconsecutive levels with the same name. However, as one final example, let's look at what onshape DOES allow.
+
+```ts
+// create a sketch with ID `rect1ID`
+const rect1ID = id+"r"+1
+createRect(rect1ID); 
+
+// create a sketch with ID `rect2ID`
+const rect2ID = id+"r"+2
+createRect(rect2ID); 
+
+// extrude the sketch identified by `rect1ID`
+const extrude1ID = id+"e"+1
+extrude(extrude1ID, rect1ID); 
+
+// extrude the sketch identified by `rect2ID`
+const extrude2ID = id+"e"+2
+extrude(extrude2ID, rect2ID); 
+```
+
+<center markdown>![](../images/featurestudio/featurestudio-ex3.gif){width=80%}</center>
+
+Onshape allows this even though the "r" level is repeated because they are consecutive, unlike the previous example. Notice how previously the order was id->r, id->e, id->r, id->e, whereas this one was id->r, id->r, id->e, id->e.
+
+### Sketches
+
+To create a sketch, use the `newSketch(context, "id", {"face": queryOfPlanarFace})`, or if you only have a `Plane` object, use `newSketch(context, "id", {"face": planeObject})`. Planes can be created with the `plane(origin, normal, x)` function. To draw on the plane use functions prefixed with `sk`, such as:
```

Comment:
People should be using newSketchOnPlane most of the time, newSketch is the Onshape sketch feature which is overkill, it's also query based which is usually less helpful